packages/components/src/dropdown-menu-v2-ariakit/test/index.tsx

```diff
@@ -0,0 +1,913 @@
+/**
+ * External dependencies
+ */
+import { render, screen, waitFor } from '@testing-library/react';
+import { press, click, hover, type } from '@ariakit/test';
```

Comment:
With @diegohaz we came to the conclusion that `@testing-library/user-event`'s way of simulating user events (especially key presses) just doesn't work well with ariakit (likely it's related to flushing microtasks). Using `@ariakit/test` instead seems to work well and allows us to write unit tests without any hacky workarounds.

Reply:
Yes, just bear in mind that `@ariakit/test` isn't as stable as `@ariakit/react`. You can expect more breaking changes without any deprecation warnings (but it will be described in the changelog).